### DISCUSSION TITLE
Is there a way to use a pre-fitted classifier?

### DISCUSSION THREAD

**Discussion by orshur** [scikit-activeml/scikit-activeml#472] [Q&A] [answered]
I'm trying to understand how to utilize the stream querying strategies properly. As far as I can tell, all the examples in the documentation start by fitting a SklearnClassifier. 

Is it possible to use an already fitted sklearn classifier? Currently attempting this with the following minimal example does not work :
```
import numpy as np
from skactiveml.stream import FixedUncertainty
from skactiveml.classifier import SklearnClassifier
from sklearn.linear_model import LogisticRegression

lr = LogisticRegression()
lr.fit([[1, 2], [1, 3]], [1, 0])

clf = SklearnClassifier(lr, classes=np.array([0, 1]))

qs = FixedUncertainty(budget=0.2)

X = np.array([[1, 2], [1, 3]])
y = np.array([1, 0])

for xt, yt in zip(X, y):
    sampled_indices, utilities = qs.query(candidates=xt.reshape([1, -1]), clf=clf, return_utilities=True)  # will throw NotFittedError 
    print(sampled_indices, utilities)
```

**Comment by tpham93** [comment, marked as answer]
Dear @orshur,

If you have a pre-trained model and no access to the prior training data, or it's not feasible to retrain it from scratch, it's currently not easily possible with the available SklearnClassifier wrapper. It is a topic worth looking into with the popularity of fine-tuning pre-trained models.

If it's about training the classifier with initial training data that is readily available before the active learning cycle starts, it is possible to wrap the classifier first and train the wrapped classifier afterward (see example below).

``` python
import numpy as np
from skactiveml.stream import FixedUncertainty
from skactiveml.classifier import SklearnClassifier
from sklearn.linear_model import LogisticRegression

lr = LogisticRegression()
clf = SklearnClassifier(lr, classes=np.array([0, 1]))
clf.fit([[1, 2], [1, 3]], [1, 0])

qs = FixedUncertainty(budget=0.2, classes=[0, 1])

X = np.array([[1, 2], [1, 3]])
y = np.array([1, 0])

for xt, yt in zip(X, y):
    sampled_indices, utilities = qs.query(candidates=xt.reshape([1, -1]), clf=clf, return_utilities=True) 
    print(sampled_indices, utilities)
```

**Reply by orshur** [reply]
Thanks for clarifying @tpham93 !
Yes, I was referring to the case where a pre-trained model already exists. As you mentioned, many real-world scenarios (training data not available, costly retraining, long validation process for production models, etc.) could greatly benefit from having this capability.

**Comment by mherde** [comment]
Hi @orshur 

We've just released `skactiveml 0.6.0`, where the above (updated) example with pre-trained models is supported. Feel free to reopen the discussion if you still have issues in this regard.
